seahub/api2/views.py

```diff
@@ -3573,6 +3575,35 @@ def get(self, request, repo_id):
         except IOError as e:
             return api_error(status.HTTP_500_INTERNAL_SERVER_ERROR, str(e))
 
+_REPO_ID_PATTERN = re.compile(r'[-0-9a-f]{36}')
+
+class RepoTokensView(APIView):
+    authentication_classes = (TokenAuthentication, )
+    permission_classes = (IsAuthenticated,)
+    throttle_classes = (UserRateThrottle, )
+
+    @json_response
+    def get(self, request, format=None):
+        repos = request.GET.get('repos', None)
+        if not repos:
+            return api_error(status.HTTP_400_BAD_REQUEST, "You must specify libaries ids")
+
+        repos = [repo for repo in repos.split(',') if repo]
+        if any([not _REPO_ID_PATTERN.match(repo) for repo in repos]):
+            return api_error(status.HTTP_400_BAD_REQUEST, "Libraries ids are invalid")
+
+        if any([not seafile_api.check_repo_access_permission(
```

Comment:
That could be a problem. The same may also happen when a library is deleted. 

I can think of two options here:

- Option 1: only return tokens for repos the user can access, and simply ignore those the user can't access. Dirty and quick.
- Option 2: tell the client the server failed to access some repos, like this:

```python
{
    'ok': {
        'repo_1': 'token_1'
        'repo_2': 'token_2'
    },
    'failed': {
        'repo_3': 'access denied',
        'repo_4': 'no such repo',
    }
}
```